src/snowflake/snowpark/modin/plugin/_internal/snowpark_pandas_types.py

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import datetime
+import inspect
+from abc import ABCMeta, abstractmethod
+from typing import Any, Callable, Optional, Union
+
+import numpy as np
+import pandas as native_pd
+
+from snowflake.snowpark.modin.plugin.utils.warning_message import WarningMessage
+from snowflake.snowpark.types import DataType, LongType
+
+TIMEDELTA_WARNING_MESSAGE = (
+    "Snowpark pandas support for Timedelta is still under development."
```

Comment:
Is this something we've already promised to customers?  Otherwise would leave more generic.
```suggestion
    "Snowpark pandas support for Timedelta is not currently available."
```